api/src/models/project-view.ts

```diff
@@ -39,6 +39,41 @@ export class GetProjectData {
   }
 }
 
+/**
+ * Pre-processes GET /projects/{id} public (published) project data
+ *
+ * @export
+ * @class GetPublicProjectData
+ */
+export class GetPublicProjectData {
```

Comment:
I'm thinking it might be nice to separate the public models from the non-public models.

Maybe put them all under `api/src/models/public`.